### PR TITLE
fix(docs): Add an example for the historization of items

### LINKED ISSUE #709: fix(docs): Add an example for the historization of items
Following https://github.com/probabl-ai/skore/issues/504, we now have historization of items. This new feature should be put forward with an example in the documentation.

### PR DESCRIPTION
**Pull request by sylvaincom**
Will close #709

### CODE REVIEW

File: examples/plot_04_tracking_items.py [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+# Creating and loading the skore project
+# ======================================
+
+# %%
+
+# remove the skore project if it already exists
+subprocess.run("rm -rf my_project_track.skore".split())
+
+# create the skore project
+subprocess.run("python3 -m skore create my_project_track".split())
```
Comment (thomass-dev) [inline examples/plot_04_tracking_items.py:36]: Please can you use tmp directory, as in other examples?

File: examples/plot_04_tracking_items.py [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+
+import matplotlib.image as mpimg
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import plotly.express as px
+from sklearn import datasets
+from sklearn.linear_model import Lasso
+
+import skore
```
Comment (thomass-dev) [inline examples/plot_04_tracking_items.py:24]: Please can you move the imports to the cells that use them, as in other examples?

File: examples/plot_04_tracking_items.py (hunk 1 above)
Comment (sylvaincom) [reply examples/plot_04_tracking_items.py]: Done

File: examples/plot_04_tracking_items.py (hunk 2 above)
Comment (sylvaincom) [reply examples/plot_04_tracking_items.py]: Done

File: examples/plot_04_tracking_items.py [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+df_track = pd.DataFrame(
+    {
+        "primitive": list_primitive,
+        "created_at": list_created_at,
+        "updated_at": list_updated_at,
+    }
+)
+df_track.insert(0, "iteration_number", np.arange(len(df_track)))
+df_track["created_at"] = pd.to_datetime(df_track["created_at"])
+df_track["updated_at"] = pd.to_datetime(df_track["updated_at"])
```
Comment (thomass-dev) [inline examples/plot_04_tracking_items.py:101]: These lines cause the CI to failed: 

```
../examples/plot_04_tracking_items.py unexpectedly failed to execute correctly:

    Traceback (most recent call last):
      File "/home/runner/work/skore/skore/examples/plot_04_tracking_items.py", line 136, in <module>
        fig.write_image("plot_04_track.png", scale=4)
      File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/plotly/basedatatypes.py", line 3835, in write_image
        return pio.write_image(self, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/plotly/io/_kaleido.py", line 266, in write_image
        img_data = to_image(
                   ^^^^^^^^^
      File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/plotly/io/_kaleido.py", line 143, in to_image
        img_bytes = scope.transform(
                    ^^^^^^^^^^^^^^^^
      File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/kaleido/scopes/plotly.py", line 233, in transform
        img = kaleido.to_image_block(spec, Path(self._tempfile.name).absolute(), self._topojson, self._mapbox_access_token, debug=debug, tmp_path=self._tmp_path)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/kaleido/__init__.py", line 43, in to_image_block
        return asyncio.run(to_image(spec, f, topojson, mapbox_token, debug=debug, tmp_path=tmp_path))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/asyncio/runners.py", line 194, in run
        return runner.run(main)
               ^^^^^^^^^^^^^^^^
      File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/asyncio/runners.py", line 118, in run
        return self._loop.run_until_complete(task)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/asyncio/base_events.py", line 687, in run_until_complete
        return future.result()
               ^^^^^^^^^^^^^^^
      File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/kaleido/__init__.py", line 129, in to_image
        response = await tab.send_command("Runtime.callFunctionOn", params=params)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/concurrent/futures/thread.py", line 58, in run
        result = self.fn(*self.args, **self.kwargs)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/choreographer/pipe.py", line 54, in write_json
        message = json.dumps(obj, ensure_ascii=False, cls=self.cls)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/json/__init__.py", line 238, in dumps
        **kw).encode(obj)
              ^^^^^^^^^^^
      File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/json/encoder.py", line 200, in encode
        chunks = self.iterencode(o, _one_shot=True)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/json/encoder.py", line 258, in iterencode
        return _iterencode(o, 0)
               ^^^^^^^^^^^^^^^^^
      File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/choreographer/pipe.py", line 32, in default
        return json.JSONEncoder.default(self, obj)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/json/encoder.py", line 180, in default
        raise TypeError(f'Object of type {o.__class__.__name__} '
    TypeError: Object of type datetime is not JSON serializable
```

Comment (sylvaincom) [reply examples/plot_04_tracking_items.py]: Done

**Comment by sylvaincom** [comment]
I got a `AttributeError: module 'skore' has no attribute 'create'`

File: examples/plot_04_tracking_items.py [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+temp_dir = tempfile.TemporaryDirectory(prefix="skore_example_")
+temp_dir_path = Path(temp_dir.name)
+
+# %%
+import subprocess
+
+# create the skore project
+subprocess.run(
+    f"python3 -m skore create my_project --working-dir {temp_dir.name}".split()
+)
```
Comment (thomass-dev) [inline examples/plot_04_tracking_items.py:32]: please use `skore.create` after merge of https://github.com/probabl-ai/skore/pull/744

**Comment by thomass-dev** [comment]
> I got a `AttributeError: module 'skore' has no attribute 'create'`, does someone know what is going on?

The CI succeed so i suppose you talk about your local tests? Please pull latest changes and re-install skore.

File: examples/plot_04_tracking_items.py (hunk 4 above)
Comment (sylvaincom) [reply examples/plot_04_tracking_items.py]: Done

File: examples/plot_04_tracking_items.py [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+
+# %%
+# Let us store several integer values for a same item called ``my_int``, each storage
+# being separated by 0.1 second:
+
+# %%
+import time
+
+my_project.put("my_int", 4)
+time.sleep(0.1)
```
Comment (thomass-dev) [inline examples/plot_04_tracking_items.py:42]: What's the purpose of this sleep?

**Review by MarieS-WiMLDS — COMMENTED** [review]
I have one question and one comment:
1. For now, this doesn't document the tree button we have in skore UI. Is it intentional?
2. It feels a bit long, could we skip the part about the updated date and focus on the value update, which in my opinion is what provides value to the user?

**Review by thomass-dev — APPROVED** [review]

**Comment by sylvaincom** [comment]
@MarieS-WiMLDS I created another ticket (https://github.com/probabl-ai/skore/issues/754) so that a v0 example on tracking items exists in the documentation, then we are can iterate

**Pull request merged**